### PR TITLE
Add transaction management

### PR DESCRIPTION
**Pull request by gorshkovayulia**
I don't think we need transaction in GET method

### CODE REVIEW

File: src/main/java/com/example/dog/dao/JdbcDogDAO.java [hunk 1]
```diff
@@ -48,60 +48,78 @@ public Dog get(int id) {
...
+            conn = dataSource.getConnection();
+            conn.setAutoCommit(false);
             statement = conn.prepareStatement(SQL_UPDATE_DOG);
             statement.setString(1, dog.getName());
             statement.setInt(2, dog.getHeight());
             statement.setInt(3, dog.getWeight());
             statement.setString(4, getDateString(dog.getDateOfBirth()));
             statement.setInt(5, id);
             int affectedRows = statement.executeUpdate();
+            conn.commit();
```
Comment (ctapobep) [inline src/main/java/com/example/dog/dao/JdbcDogDAO.java:95]: Shouldn't you roll back if the exception happens on the next line?

File: src/main/java/com/example/dog/dao/JdbcDogDAO.java [hunk 2]
```diff
@@ -48,60 +48,78 @@ public Dog get(int id) {
...
             if (affectedRows == 0) {
                 throw new ObjectNotFoundException("Dog with id=" + id + " was not found!");
             }
             updatedDog = get(id);
         } catch (SQLException e) {
-            throw new RuntimeException(e);
+            handleSQLException(conn, e);
```
Comment (ctapobep) [inline src/main/java/com/example/dog/dao/JdbcDogDAO.java:102]: Will `finally` work you if you throw an exception from here?

**Comment by ctapobep** [comment]
> I don't think we need transaction in GET method

This should be put as a comment in the code then. But it's better to have a tx for read-only operations as well: https://stackoverflow.com/questions/13539213/why-do-i-need-transaction-in-hibernate-for-read-only-operations/13540484#13540484

File: src/main/java/com/example/dog/dao/JdbcDogDAO.java (hunk 2 above)
Comment (gorshkovayulia) [reply src/main/java/com/example/dog/dao/JdbcDogDAO.java:102]: I'd say - yes, even if `catch` block throws an exception

File: src/main/java/com/example/dog/dao/JdbcDogDAO.java (hunk 1 above)
Comment (gorshkovayulia) [reply src/main/java/com/example/dog/dao/JdbcDogDAO.java:95]: Yep, good point

File: src/main/java/com/example/dog/dao/JdbcDogDAO.java [hunk 3]
```diff
@@ -26,12 +27,16 @@ public JdbcDogDAO(DriverManagerDataSource dataSource) {
         this.dataSource = dataSource;
     }
 
+    @Transactional(readOnly = true)
```
Comment (ctapobep) [inline src/main/java/com/example/dog/dao/JdbcDogDAO.java]: That's too early - this annotation won't work for now.

**Comment by ctapobep** [comment]
+2

File: src/main/java/com/example/dog/dao/JdbcDogDAO.java (hunk 3 above)
Comment (gorshkovayulia) [reply src/main/java/com/example/dog/dao/JdbcDogDAO.java]: oops, forgot to delete it

**Pull request merged**